### PR TITLE
Support Redirects

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
Usually, VTEX customers need to add custom redirects to their store. This PR adds this first-party capability by making `gatsby-source-vtex` receive a `getRedirects` function. This `async function` receives no argument and must return a list of Redirects. This way, our customer may fetch these redirects list from anywhere he wants.

## How it works? 
Just take a look in the attached PRs:
https://github.com/vtex-sites/btglobal.store/pull/270
https://github.com/vtex-sites/marinbrasil.store/pull/400
https://github.com/vtex-sites/storecomponents.store/pull/565

## How to test it?
<em>Describe the steps with bullet points. Is there any external link that can be used to better test it or an example?</em> 

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: packages/gatsby-source-vtex/src/redirects.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      isPermanent: type === 'PERMANENT',
+      statusCode: type === 'PERMANENT' ? 301 : 302,
+    })
+  }
+
+  return redirects
+}
+
+const getRedirects = async () => {
+  const maybeRedirects = await redirectsFromCSV({ delimiter: ';' })
```
Comment (igorbrasileiro) [inline packages/gatsby-source-vtex/src/redirects.ts:70]: The default delimiter is the comma, maybe the user would want to choose which delimiter he wants to use.

File: packages/gatsby-source-vtex/src/gatsby-node.ts [hunk 2]
```diff
@@ -213,16 +215,35 @@ export const sourceNodes: GatsbyNode['sourceNodes'] = async (
...
+  }: Options
 ) => {
   /**
    * Create all proxy rules for VTEX Store
    * If adding a new rule, don't forget to modify ./gatsby-config.ts dev proxy
    * so it also works in dev mode
    */
 
+  const activity = reporter.activityTimer(
+    '[gatsby-source-vtex]: soucing Redirects'
```
Comment (igorbrasileiro) [inline packages/gatsby-source-vtex/src/gatsby-node.ts:234]: Did you mean source?

```suggestion
    '[gatsby-source-vtex]: sourcing Redirects'
```

File: packages/gatsby-source-vtex/src/redirects.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import fs from 'fs'
```
Comment (igorbrasileiro) [inline packages/gatsby-source-vtex/src/redirects.ts:1]: Maybe you need lint this file

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]

File: packages/gatsby-source-vtex/src/redirects.ts (hunk 3 above)
Comment (tlgimenes) [reply packages/gatsby-source-vtex/src/redirects.ts:1]: my `linter`, and `git staged` are not complaining about this file. What's your linter is complaining about ?

File: packages/gatsby-source-vtex/src/redirects.ts (hunk 1 above)
Comment (tlgimenes) [reply packages/gatsby-source-vtex/src/redirects.ts:70]: Sure, if the user wants to use a different delimiter, he can pass a different getRedirects function altogether :) 

Also, if the user doesn't want to use a csv, he can fetch these redirects from an api etc

Comment (igorbrasileiro) [reply packages/gatsby-source-vtex/src/redirects.ts]: Nice!

**Review by igorbrasileiro — APPROVED** [review]
LGTM

**Review by emersonlaurentino — APPROVED** [review]

**Pull request merged**